openwisp_radius/base/models.py

```diff
@@ -197,7 +198,25 @@ def clean(self):
             )
 
 
-class AbstractRadiusCheck(OrgMixin, AutoUsernameMixin, TimeStampedEditableModel):
+class UniqueAtrributeMixin(object):
+    def clean(self):
+        """
+        checks if the check or reply attribute is unique
+        """
+        model = type(self).__name__
+        if 'RadiusGroup' in model:
+            options = dict(group=self.group)
+        else:
+            options = dict(organization=self.organization, user=self.user)
```

Comment:
Should I separate this conditional logic into separate mixin for groups?

Reply:
This can be returned by a method, you can then create two mixins, one for group check/replies and one for non group check/replies, which return the right value.

eg:

```python
GroupAttributeValidationMixin(AttributeValidationMixin):
    def __get_validation_queryset_kwargs(self):
        return dict(group=self.group)
```